### PR TITLE
feat: default oauth scopes

### PR DESCRIPTION
**Pull request by MichaelGoberling**
## Description

Add default scopes to oauth step, so users can add I/O Management API to their credential and skip scope config

Scope configuration is still possible, but with this it'd be optional

(Note: default scopes are already supported with the deprecated jwt step, this is bringing oauth up to par with that)

## Related Issue

[ACNA-2762](https://jira.corp.adobe.com/browse/ACNA-2762)

## How Has This Been Tested?
`npm run test`

[Real test using default scopes ](https://github.com/MichaelGoberling/app-builder-gh-action/actions/runs/7729367137/job/21072354956)

[Real test using configured scopes](https://github.com/MichaelGoberling/app-builder-gh-action/actions/runs/7729340138/job/21072261918)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -88,7 +102,9 @@ async function generateOAuthSTSAuthToken (params = {}) {
     technical_account_email: techAccEmail,
     technical_account_id: techAccId,
     ims_org_id: imsOrgId,
-    scopes: scopes.split(',')
+    scopes: [
```
Comment (purplecabbage) [inline src/utils.js:105]: Why so tricky? Can't this just be `scopes: finalScopes` ?
or `finalScopes.slice()` ?

**Review by purplecabbage — APPROVED** [review]

**Review by shazron — APPROVED** [review]

File: src/utils.js (hunk 1 above)
Comment (MichaelGoberling) [reply src/utils.js:105]: Originally I just wrote `scopes: finalScopes`, but then I saw below in the auth step it was written this way and decided to keep things consistent. Should work either way, but maybe there's a hidden benefit to using the spread operator? Any idea @shazron? I think this was written when you found the double array bug

File: src/utils.js [hunk 2]
```diff
@@ -71,14 +71,28 @@ async function generateOAuthSTSAuthToken (params = {}) {
...
+    required: ['clientId', 'clientSecret', 'techAccId', 'techAccEmail', 'imsOrgId']
   }
 
   const { valid, errors } = validate(schema, params)
   if (!valid) {
     throw new Error(`[generateOAuthSTSAuthToken] Validation errors: ${JSON.stringify(errors, null, 2)}`)
   }
 
+  const finalScopes = scopes
+    ? scopes.split(',')
```
[the remark this answers is not in the document]
Comment (MichaelGoberling) [reply src/utils.js:83]: `scopes.split(',')` will always return an array, right? Even if the string is empty or a string with no comma

File: src/utils.js (hunk 1 above)
Comment (MichaelGoberling) [reply src/utils.js:105]: Going to keep this as is for now, IIWIT

**Pull request merged**